### ISSUE TITLE
RIOT-OS support

### ISSUE THREAD

**Issue by cselti** [platformio/platformio-core#3456]
Supporting [RIOT-OS](https://www.riot-os.org/) as a framework could be useful. It supports several platforms (ESP, ATMega, ARM, ...), several sensor modules (as [drivers](https://github.com/RIOT-OS/RIOT/tree/master/drivers)), etc... 
I don't know too much about the RIOT-OS, but I think it could be fit in the PlatformIO's frameworks (probably a similar way like to the Zephyr project?)
I have checked repo of the RIOT - I found a [former ticket](https://github.com/RIOT-OS/RIOT/issues/8591) about the topic, I think the RIOT developers don't plan to integration at the moment.

**Comment by ivankravets** [comment]
Please upvote 👍 this feature request (smile in the right upper corner of your comment). We work on the most requested features by our Community. Otherwise, @gebart and his team can provide integration from their side.

/cc @Citrullin  @basilfx @douardda @andrea-nisti

**Comment by Citrullin** [comment]
@ivankravets I thought about adding the support as well, because I worked with platformIO on something work related. :) So, feel free to ping me at any point, if you need help. Would like to understand platformIO even more :)

**Comment by Dahkai** [comment]
Good luck 🤞

**Comment by akshaim** [comment]
Hope this materialize soon !

**Comment by aabadie** [comment]
Any progress @Citrullin ?

**Comment by Citrullin** [comment]
> Any progress @Citrullin ?

I haven't worked on it. So, you probably should ask @ivankravets

**Comment by dzarda** [comment]
@ivankravets Perhaps could be viewed as *Makefile project support*? I.e. ChibiOS has a thoroughly Makefile-based structure and would also benefit from it.

**Comment by leonty** [comment]
@ivankravets Hello, thanks again for a great tool! Do you guys still consider adding support of the RIOT OS? It's really a serious OS with support of many SoCs (>40) and boards (>200). Still it has very simple build system based on makefiles. There are 114 upvotes already)

**Comment by ivankravets** [comment]
Hi there,
The RIOT-OS team works on integration. Stay tuned! 🥳

**Comment by mderouss** [comment]
Great that this is happening ! I've looked at RIOT for a few projects in the last 18 months, but have gotten used to PlatformIO :)  Will certainly encourage me to use RIOT.

**Comment by mansk789** [comment]
any news for this topic? It would be great if there platformio would support RIOT-OS

**Comment by leonty** [comment]
There was an open issue in the RIOT OS repo, but now it's closed, no one seemed to be interested in this integration...
https://github.com/RIOT-OS/RIOT/issues/8591

**Comment by Citrullin** [comment]
> 
> 
> There was an open issue in the RIOT OS repo, but now it's closed, no one seemed to be interested in this integration...
> [RIOT-OS/RIOT#8591](https://github.com/RIOT-OS/RIOT/issues/8591)

Yes, I closed it, since it isn't so much of an issue in RIOT itself. I still would like to see it :)

**Comment by leonty** [comment]
@ivankravets Please, give us some details if any on the work progress.

**Comment by smurfix** [comment]
Is anybody interested in doing the work? I seem to need it and would contribute some $$$ to the effort.

**Comment by Citrullin** [comment]
> Is anybody interested in doing the work? I seem to need it and would contribute some $$$ to the effort.

I added it to the [RIOT OS forum](https://forum.riot-os.org/t/platformio-integration/3590). Maybe you can add some details there. It's the best place to reach out to the RIOT OS users.

**Comment by bitsformbyte** [comment]
Is there anyone working on this? @ivankravets

**Comment by redradist** [comment]
Some updates on it ??

**Comment by Citrullin** [comment]
> Some updates on it ??

Sadly not. I would love to work on it or fund it eventually, but it's not a priority right now. 
Also haven't heard of anyone in the RIOT OS community working on it.
